### PR TITLE
Add NetBSD-generic32, which is BSD-generic32 but also -latomic.

### PR DESCRIPTION
**Pull request by he32**

### CODE REVIEW

File: Configurations/10-main.conf [hunk 1]
```diff
@@ -1018,6 +1018,10 @@ my %targets = (
         shared_target    => "bsd-gcc-shared",
         shared_cflag     => "-fPIC",
     },
+    "NetBSD-generic32" => {
+	inherit_from	=> [ "BSD-generic32" ],
+	ex_libs		=> add(threads("-latomic")),
```
Comment (mspncp) [inline Configurations/10-main.conf:1023]: It looks like something is wrong here with the indentation?

Comment (he32) [reply Configurations/10-main.conf]: I submitted an update converting tabs -> spaces.

**Review by hlandau — APPROVED** [review]

**Comment by paulidale** [comment]
We'll also need a community maintainer for NetBSD because this looks like a new / unsupported platform.

**Comment by he32** [comment]
> We'll also need a community maintainer for NetBSD because this
> looks like a new / unsupported platform.

Oh, dear.

What do I / we need to do?

The NetBSD project has ~forever carried around and integrated a
version of OpenSSL in its source tree, and we have recently
started to transition from 1.1.1t to 3.0.8.  Ref.

  https://github.com/NetBSD/src/tree/trunk/crypto/external/bsd/openssl

I'm not the one doing this integration work, but I do have the
"NetBSD commit bit".  I probably want to bring in the guy doing
the work.

Best regards,

- Håvard

**Comment by paulidale** [comment]
To become the community maintainer, submit a pull request adding a line to the list:

https://github.com/openssl/general-policies/blob/master/policy-supplemental/platforms.md


It would also be prudent to read [what's required](https://github.com/openssl/general-policies/blob/master/policies/platform-policy.md#community).  It's not onerous, basically just a commitment to assist the core team if there is a problem with the platform.

**Review by paulidale — APPROVED** [review]
I'm okay with this going in under a trivial tag.

The make this as trivial you need to amend each commit and add a line "CLA: trivial" down the bottom.  Then force push.

**Comment by tom-cosgrove-arm** [comment]
@paulidale Note that when another contributor raised [a PR to add updated ios simulator config targets](https://github.com/openssl/openssl/pull/21309), which was also considered _acceptable as trivial_, [@mattcaswell wrote](https://github.com/openssl/openssl/pull/21309#issuecomment-1611463624)

> Our policy is to not add new configuration targets without having a volunteer to maintain the platform. Anyone may volunteer by opening a PR to create an entry for the target in the "Community platforms" section of this page:

> https://github.com/openssl/general-policies/blob/master/policy-supplemental/platforms.md

**Pull request closed without merging**